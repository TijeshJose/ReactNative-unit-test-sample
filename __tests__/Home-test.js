import 'react-native';
import React from 'react';
import Home from '../src/home';

import renderer from 'react-test-renderer';
import AsyncStorage from '@react-native-community/async-storage';

//check api response if api implemented in home page
global.fetch = jest.fn().mockImplementation((url, data) => {
    var p = new Promise((resolve, reject) => {
        resolve({
            json: function () {
                return {
                    "status": true
                }
            }
        })
    })
    return p;
});


//check home page renders correctly
it('Home page renders correctly', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
});

//check with mock async storage value
it('checks if Async Storage is used', async () => {
    await AsyncStorage.getItem('email');

    expect(await AsyncStorage.getItem).toBeCalledWith('email');
})