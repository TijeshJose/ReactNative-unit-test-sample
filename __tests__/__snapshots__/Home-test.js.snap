// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
      "padding": 20,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 24,
          "fontWeight": "bold",
          "marginTop": 20,
        }
      }
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#000",
        "borderRadius": 5,
        "justifyContent": "center",
        "marginTop": 20,
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 16,
          "fontWeight": "400",
        }
      }
    >
      Logout
    </Text>
  </View>
</View>
`;
